src/main/groovy/edu/oregonstate/mist/metaxe/XEAppDAO.groovy

```diff
@@ -0,0 +1,123 @@
+package edu.oregonstate.mist.metaxe
+
+import com.fasterxml.jackson.core.JsonParseException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.JsonMappingException
+import javax.ws.rs.core.UriBuilder
+
+@groovy.transform.TypeChecked
+class XEAppDAO {
+    private URI esUrl
+    private ObjectMapper mapper = new ObjectMapper()
+
+    XEAppDAO(URI elasticsearchUrl) {
+        this.esUrl = elasticsearchUrl
+    }
+
+    ESResult getById(String id) {
+        def builder = UriBuilder.fromUri(this.esUrl)
+        def url = builder.path('{id}').build(id).toURL()
+
+        println "get by id: ${url}"
+
+        InputStream jsonStream
+        try {
+            jsonStream = url.newInputStream()
+        } catch (FileNotFoundException e) {
+            // 404 -> return null
+            return null
+        } catch (IOException e) {
+            // some other http error
+            throw new ElasticsearchException(url, e)
+        }
+
+        // Catch JsonMappingException and JsonParseException here,
+        // as otherwise jersey will catch them and interpret them
+        // as an error in the client request
+        //
+        // Catch IOException as well, because why not
+        try {
+            mapper.readValue(jsonStream, ESResult) // return
+        } catch (IOException e) {
+            // read error
+            throw new ElasticsearchException(url, e)
+        } catch (JsonMappingException e) {
+            throw new ElasticsearchException(url, e)
+        } catch (JsonParseException e) {
+            throw new ElasticsearchException(url, e)
+        }
+    }
+
+    ESHits search(
+        String q, String instance, String version,
```

Comment:
This is my favorite style for splitting long parameter lists. I've changed it to match the rest of the code.